Src/FluentAssertions/DataRowCollectionAssertionExtensions.cs

```diff
@@ -0,0 +1,347 @@
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataRow>> ContainEquivalentOf(this GenericCollectionAssertions<DataRow> assertion, DataRow expected, string because = "",
+            params object[] becauseArgs)
+        {
+            return ContainEquivalentOf(assertion, expected, config => config, because, becauseArgs);
+        }
+
+        /// <summary>
+        /// Asserts that a <see cref="DataRowCollection"/> contains at least one object equivalent to another object.
+        /// </summary>
+        /// <remarks>
+        /// See the documentation for <see cref="DataRowAssertions{TDataRow}"/> for information about when <see cref="DataRow"/>
+        /// objects are considered equivalent.
+        /// </remarks>
+        /// <param name="expected">The expected <see cref="DataRow"/>.</param>
+        /// <param name="config">
+        /// A reference to the <see cref="EquivalencyAssertionOptions{DataRow}"/> configuration object that can be used
+        /// to influence the way the object graphs are compared. You can also provide an alternative instance of the
+        /// <see cref="EquivalencyAssertionOptions{DataRow}"/> class. The global defaults are determined by the
+        /// <see cref="AssertionOptions"/> class.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataRow>> ContainEquivalentOf(this GenericCollectionAssertions<DataRow> assertion, DataRow expected, Func<EquivalencyAssertionOptions<DataRow>,
+                EquivalencyAssertionOptions<DataRow>> config, string because = "", params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(config, nameof(config));
+
+            if (assertion.Subject is null)
+            {
+                Execute.Assertion
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith("Expected {context:collection} to contain equivalent of {0}{reason}, but found <null>.", expected);
+            }
+
+            EquivalencyAssertionOptions<DataRow> options = config(AssertionOptions.CloneDefaults<DataRow>());
+            IEnumerable<object> actualItems = assertion.Subject.Cast<object>();
```

Comment:
Why casting to `object` instead of `DataRow`?

Reply:
I believe copy/pasted this code from the old non-generic collection support, where it was casting `IEnumerable` to `IEnumerable<T>`. I think actually no cast is necessary since in this method, `Subject` is already an `IEnumerable<T>` (where `T` == `DataRow`).